mbed-client-libservice/ns_types.h

```diff
@@ -117,7 +117,7 @@ typedef int_fast32_t int_fast24_t;
 
 /* C11's "alignas" macro, emulated for integer expressions if necessary */
 #ifndef __alignas_is_defined
-#if defined __CC_ARM || defined __TASKING__
+#if defined __CC_ARM || defined __TASKING__ || ( defined(__ARMCC_VERSION) && __ARMCC_VERSION >= 6000000 )
```

Comment:
So ARM C 6 doesn't define __CC_ARM? But ARM C 5 does define __ARMCC_VERSION, doesn't it, so you could just check for that?

Reply:
Or check for `__clang__` - would make more sense as you're looking for a language front-end feature.
Although does it actually support __align? Can't see it in documentation.

Reply:
Whereas it does support the GNU C attribute, and it would pick it up without this change, as `__GNUC__` is defined.